main/src/ca/uwaterloo/flix/language/phase/Parser2.scala

```diff
@@ -2889,6 +2889,36 @@ object Parser2 {
       close(mark, TreeKind.Expr.FixpointWhere)
     }
 
+    private def predicateShape()(implicit s: State): Mark.Closed = {
+      val mark = open()
+      nameUnqualified(NAME_PREDICATE, SyntacticContext.Expr.OtherExpr)
+
+      // check for shape (_, _, _ ...)
+      if (!at(TokenKind.ParenL)) {
+        closeWithError(open(), UnexpectedToken(
+          expected = NamedTokenSet.FromKinds(Set(TokenKind.ParenL)),
+          actual = Some(nth(0)),
+          sctx = SyntacticContext.Expr.OtherExpr,
+          hint = Some("provide a predicate shape."),
```

Comment:
could the hint include the syntax? like `provide a predicate shape like '(_, _)'`